### ISSUE TITLE
error when standalone port is taken

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#3406]
When you run Certbot with the standalone plugin and the required port is taken, you see a traceback like this when the `--debug` flag is present:
```
$ sudo certbot -d example.com --standalone certonly -t --debug
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Starting new HTTPS connection (1): acme-staging.api.letsencrypt.org
Obtaining a new certificate
Performing the following challenges:
tls-sni-01 challenge for example.com

-------------------------------------------------------------------------------
The program apache2 (process ID 13565) is already listening on TCP port 80. This
will prevent us from binding to that port. Please stop the apache2 program
temporarily and then try again.
-------------------------------------------------------------------------------
Press Enter to Continue
Cleaning up challenges
Traceback (most recent call last):
  File "/usr/local/bin/certbot", line 9, in <module>
    load_entry_point('certbot', 'console_scripts', 'certbot')()
  File "/home/bmw/src/certbot/certbot/main.py", line 770, in main
    return config.func(config, plugins)
  File "/home/bmw/src/certbot/certbot/main.py", line 565, in obtain_cert
    _, action = _auth_from_domains(le_client, config, domains, lineage)
  File "/home/bmw/src/certbot/certbot/main.py", line 104, in _auth_from_domains
    lineage = le_client.obtain_and_enroll_certificate(domains)
  File "/home/bmw/src/certbot/certbot/client.py", line 275, in obtain_and_enroll_certificate
    certr, chain, key, _ = self.obtain_certificate(domains)
  File "/home/bmw/src/certbot/certbot/client.py", line 247, in obtain_certificate
    self.config.allow_subset_of_names)
  File "/home/bmw/src/certbot/certbot/auth_handler.py", line 70, in get_authorizations
    resp = self._solve_challenges()
  File "/home/bmw/src/certbot/certbot/auth_handler.py", line 108, in _solve_challenges
    resp = self.auth.perform(self.achalls)
  File "/home/bmw/src/certbot/certbot/plugins/standalone.py", line 217, in perform
    "At least one of the (possibly) required ports is "
MisconfigurationError: At least one of the (possibly) required ports is already taken.
```
However, if you specify ports with `--http-01-port`/`--tls-sni-01-port`, the traceback is:
```
$ sudo certbot -d example.com --standalone certonly -t --debug --tls-sni-01-port 5001 --http-01-port 5002
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Starting new HTTPS connection (1): acme-staging.api.letsencrypt.org
Obtaining a new certificate
Performing the following challenges:
tls-sni-01 challenge for example.com

-------------------------------------------------------------------------------
Could not bind TCP port 5001 because it is already in use by another process on
this system (such as a web server). Please stop the program in question and then
try again.
-------------------------------------------------------------------------------
Press Enter to Continue
Traceback (most recent call last):
  File "/usr/local/bin/certbot", line 9, in <module>
    load_entry_point('certbot', 'console_scripts', 'certbot')()
  File "/home/bmw/src/certbot/certbot/main.py", line 770, in main
    return config.func(config, plugins)
  File "/home/bmw/src/certbot/certbot/main.py", line 565, in obtain_cert
    _, action = _auth_from_domains(le_client, config, domains, lineage)
  File "/home/bmw/src/certbot/certbot/main.py", line 104, in _auth_from_domains
    lineage = le_client.obtain_and_enroll_certificate(domains)
  File "/home/bmw/src/certbot/certbot/client.py", line 275, in obtain_and_enroll_certificate
    certr, chain, key, _ = self.obtain_certificate(domains)
  File "/home/bmw/src/certbot/certbot/client.py", line 247, in obtain_certificate
    self.config.allow_subset_of_names)
  File "/home/bmw/src/certbot/certbot/auth_handler.py", line 70, in get_authorizations
    resp = self._solve_challenges()
  File "/home/bmw/src/certbot/certbot/auth_handler.py", line 114, in _solve_challenges
    assert len(resp) == len(self.achalls)
TypeError: object of type 'NoneType' has no len()
```
The error being raised here should be the same as the one above. I suspect this is an easy fix.

**Comment by JulianGindi** [comment]
Do you mean the error presented to the user or the actual error being thrown (`TypeError` vs `MisconfigurationError`)? I'm not sure an error with `len(res) == len(self.achalls)` should necessarily throw a `MisconfigurationError`. 

What do you think?

**Comment by bmw** [comment]
The second traceback is later in execution than the first. I think the problem should ideally be caught at the same time as it was in the first traceback causing the same error type and message to be raised.

**Comment by ClifHouck** [comment]
@bmw it looks like this has been fixed by you in a recent commit. Specifically a5df9e5a0e5fcb6a66c63e0bd02844417fdbb456

**Comment by bmw** [comment]
Fixed in #3608.
